src/torrent_info.cpp

```diff
@@ -528,7 +528,7 @@ namespace libtorrent
 		}
 		else
 		{
-			file_flags &= ~file_storage::flag_symlink;
+			file_flags &= ~(file_storage::flag_symlink & 0xffffffff);
```

Comment:
is the problem here that the enum value is signed? Since these are flags, it probably would make sense to make the enum unsigned: ``enum file_flags_t : std::uint32_t`` in file_storage.hpp